aether/sdk/drf/tests/test_views.py

```diff
@@ -0,0 +1,109 @@
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import random
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.test import RequestFactory
+from aether.sdk.unittest import UrlsTestCase
+from django.urls import reverse
+from rest_framework import status
+from aether.sdk.tests.fakeapp import models
+
+TEST_REALM = 'realm-test'
+
+
+class ViewsTest(UrlsTestCase):
+
+    def setUp(self):
+        super(ViewsTest, self).setUp()
+
+        username = 'user'
+        email = 'user@example.com'
+        password = 'secretsecret'
+        user = get_user_model().objects.create_user(username, email, password)
+
+        self.request = RequestFactory().get('/')
+        self.request.COOKIES[settings.REALM_COOKIE] = TEST_REALM
+        self.request.user = user
+
+        self.assertTrue(self.client.login(username=username, password=password))
+        self.client.cookies[settings.REALM_COOKIE] = TEST_REALM
+        test_names = ['one', 'two', 'three']
+        for i in range(3):
+            parent = models.TestModel.objects.create(name=test_names[i])
+            for _ in range(10):
+                models.TestChildModel.objects.create(
+                    name=str(random.randint(1, 100)),
+                    parent=parent,
+                )
+
+    def tearDown(self):
+        super(ViewsTest, self).tearDown()
+        models.TestModel.objects.all().delete()
+
+    def test_delete_by_filters(self):
+        org_count = models.TestChildModel.objects.count()
+        parent = models.TestModel.objects.first()
+        filtered_count = models.TestChildModel.objects.filter(parent=parent).count()
+        self.assertEqual(org_count, 30)
+        self.assertEqual(filtered_count, 10)
+
+        url = f'{reverse("testchildmodel-filtered-delete")}?parent={parent.name}'
+        self.client.delete(url)
+        self.assertEqual(
+            models.TestChildModel.objects.count(),
+            (org_count - filtered_count)
+        )
+        filtered_count = models.TestChildModel.objects.filter(parent=parent).count()
+        self.assertEqual(filtered_count, 0)
+
+    def test_update_by_filters(self):
+        org_count = models.TestChildModel.objects.count()
+        parent = models.TestModel.objects.first()
+        filtered_count = models.TestChildModel.objects.filter(parent=parent).count()
+        self.assertEqual(org_count, 30)
+        self.assertEqual(filtered_count, 10)
+        url = f'{reverse("testchildmodel-filtered-update")}?parent={parent.name}'
+        update_fields = {
+            'name': 'new-name',
+        }
+        response = self.client.patch(
+            url,
+            data=update_fields,
+            content_type='application/json'
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```

Comment:
Could you also check that the fields updates was done :grimacing: I mean, that the "children" have changed their names... :see_no_evil:

Reply:
This is still pending ;-)

Reply:
https://github.com/eHealthAfrica/aether-django-sdk-library/pull/22/files#diff-1d4b6ba20a228a6975d0a33b01a0a221R88

Reply:
:see_no_evil: Completely lost in the reviewing :woman_facepalming: